### PR TITLE
658 ctpattern

### PR DESCRIPTION
**Pull request by rowanhemsi**

### CODE REVIEW

File: cishouseholds/derive.py [hunk 1]
```diff
@@ -37,7 +38,48 @@ def substring_column(df: DataFrame, new_column_name, column_to_substr, start_pos
...
+            (0, 0, 1, "S only"),
+            (1, 1, 0, "OR+N"),
+            (1, 0, 1, "OR+S"),
+            (0, 1, 1, "N+S"),
+            (1, 1, 1, "OR+N+S"),
+        ],
+        schema=indicator_list + ["ctpattern"],
+    )
+
+    for ct_gene in ["or", "n", "s"]:
```
Comment (rowanhemsi) [inline cishouseholds/derive.py:80]: Check - are these (eg. `ct_or`) the correct variable names?

Comment (rowanhemsi) [reply cishouseholds/derive.py:80]: Also, are these variable nullable?

Comment (rowanhemsi) [reply cishouseholds/derive.py:80]: Variable names not decided - give to function as variable.
Three variables are nullable - should behave as 0

**Review by foster999 — APPROVED** [review]
Approved by group review with @howelj, @rowanhemsi, @georgefeldman  and @foster999

**Pull request merged**